[Lim Jian Yang] iP

jianyangg opened the pull request:
# Chatty
> "No wise pilot, no matter how great his talent and experience, fails to use his checklist." - Charlie Munger ([source](https://www.valuewalk.com/charlie-munger-wise-pilot/))

Chatty is your personal assistant who remembers your tasks so that you do not have to. It can
- create to-dos
- create tasks with deadlines
- create events with a start and end time
and it's _incredibly intuitive_ to use due to its ability to understand **natural language**!

Download your own personal Chatty now by going through the following steps:
1. Download the executable JAR file under "Release"
1. Run the compiled executable by doing 
```
 java -jar Duke.jar
```
Start your **free** Chatty companion now! ❗

If you are a Java programmer, you can use it to practice Java too. Here's the main method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

jianyangg committed:
add Level-0

jianyangg committed:
Level-1

jianyangg committed:
remove class files

jianyangg committed:
remove class files

jianyangg committed:
Level-2

jianyangg committed:
Level-3

jianyangg committed:
Level-4

jianyangg committed:
A-TextUiTesting

jianyangg committed:
Level-5

jianyangg committed:
Level-6

jianyangg committed:
Level-6

jianyangg committed:
Add save list feature

Task list will be saved to data/duke.txt directory

jianyangg committed:
Add load list feature

List is now loaded from data/duke.txt, if it exists.

jianyangg committed:
Merge branch 'branch-Level-7'

jianyangg committed:
Add Level-8 features to parse dates

Deadline tasks can now parse date into LocalDate objects instead of being stored as String

jianyangg committed:
Merge branch 'branch-Level-8'

jianyangg committed:
Add exception handling for invalid date string

Deadline task now catches invalid dates and ask the user to input again

jianyangg committed:
Implement more OOP principles into the code

Code was edited based on A-MoreOOP guidelines.

jianyangg committed:
Merge branch 'A-MoreOOP'

jianyangg committed:
Put all classes into one package

Done in accordance with the A-Package instructions.

jianyangg committed:
Merge branch 'A-Packages'

jianyangg committed:
Put all classes into one package

Done in accordance with the A-Package instructions.

jianyangg committed:
Merge remote-tracking branch 'origin/add-gradle-support'

jianyangg committed:
Add Gradle support to the project

Merged add-gradle-support branch with master branch

jianyangg committed:
Add JUnit tests for ToDo and Task classes

For A-JUnit task.

jianyangg committed:
Merge branch 'A-JUnit'

jianyangg committed:
Create a JAR file

JAR file should not actually be included here. It'll be released instead as v0.1

jianyangg committed:
Merge branch 'A-Jar'

jianyangg committed:
Add JavaDoc comments to the code

Completed according to the requirements specified in A-JavaDoc notes

jianyangg committed:
Add find functionality

Allows users to search for tasks using keywords.

jianyangg committed:
Refine the code to Java Coding Standard

Edited code to match Java Coding Standard

jianyangg committed:
Merge branch 'branch-A-JavaDoc'

jianyangg committed:
Merge master branch with JavaDoc and Level-9

Handled merge conflicts when merging into master from branch-A-JavaDoc and branch-Level-9

jianyangg committed:
Merge branch 'branch-A-CodingStandard'

jianyangg committed:
Implement GUI features for better user experience

Code done in alignment with Level-10 GUI guidelines

jianyangg committed:
Merge branch 'branch-Level-10'

sudarshan2401 commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,44 @@
+ * representation of start and end.
+ */
+public class Event extends Task {
+    private String fromString;
+    private String toString;
```
Informative naming convention!

jianyangg replied:
Thank you.

sudarshan2401 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,104 @@
+     * @param i Index of the task of interest.
+     * @return String representation of the task at the specified index with numbering.
+     */
+    public String taskToString(int i) {
```
nit: perhaps could change the name to prevent confusion with the toString method!

jianyangg replied:
Got it. Will do!

sudarshan2401 reviewed (commented):
LGTM! just a few nits!

Ken-Lai commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,169 @@
+
+        switch (command) {
+        case "bye":
+            return ui.sendFarewell();
```
Perhaps you can add a ```//Fallthrough``` comment to explain why break is not needed?

jianyangg replied:
That's a good point! I will do that.

Ken-Lai commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,169 @@
+        case "bye":
+            return ui.sendFarewell();
+        case "list":
+            return ui.printTaskList(tasks.toString());
```
Refer to above

Ken-Lai commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,169 @@
+            taskIndex = Integer.parseInt(inputString.split(" ")[1]) - 1;
+            // tasks.get(taskIndex).markAsDone();
+            tasks.markTaskAsDone(taskIndex);
+            return ui.markTaskAsDoneMessage(tasks.taskToString(taskIndex));
```
Refer to above

Ken-Lai commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,169 @@
+        case "unmark":
+            taskIndex = Integer.parseInt(inputString.split(" ")[1]) - 1;
+            tasks.unmarkTaskAsDone(taskIndex);
+            return ui.unmarkTaskAsDoneMessage(tasks.taskToString(taskIndex));
```
Refer to above

Ken-Lai commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,169 @@
+                String taskName = inputString.substring(5);
+                Task newTask = new ToDo(taskName);
+                tasks.addToTaskList(newTask);
+                return ui.addTaskOutputText(newTask, tasks.size());
```
Refer to above

Ken-Lai commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,169 @@
+                String to = inputString.substring(inputString.indexOf("/to") + 4);
+                Task newTask = new Event(taskName, from, to);
+                tasks.addToTaskList(newTask);
+                return ui.addTaskOutputText(newTask, tasks.size());
```
Refer to above

Ken-Lai commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,169 @@
+            taskIndex = Integer.parseInt(inputString.split(" ")[1]) - 1;
+            String msg = ui.printDeleteMessage(tasks.taskToString(taskIndex), taskIndex, tasks.size());
+            tasks.deleteTask(taskIndex);
+            return msg;
```
Refer to above

Ken-Lai commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,169 @@
+                    outputString += tasks.taskToString(i) + "\n";
+                }
+            }
+            return ui.printTaskList(outputString);
```
Refer to above

Ken-Lai commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,169 @@
+            return ui.printTaskList(outputString);
+        default:
+            return ui.showErrorMessage(
+                    "\t☹ OOPS!!! I'm sorry, but I don't know what that means. Try again using either mark <index>, unmark <index>, todo <task>, deadline <task /by ..>, event <task /from .. /to ..>, or bye.");
```
Maybe you can use line wapping to avoid exceeding the line length?

jianyangg replied:
Rights, I missed this out. Will fix this!

Ken-Lai reviewed (commented):
Overall code is good, just need a bit explanation for cases and not to exceed word limit for each row. Keep up the good work!

jianyangg reviewed (commented):
Replying to reviewers

jianyangg committed:
Improve the code quality in accordance with the CS2103T textbook

Code quality were improved with respect to areas such as comments,

line spacing to separate groups of related statements,

and avoided complicated expressions by breaking them down into simpler logic blocks

jianyangg committed:
Add assertions where appropriate to validate function arguments

The addition of these assertions should help us validate our assumptions,

thus making it easier for errors and bugs to be detected.

As there are not much logic involved in this project, focus is mainly placed

on validating the input to the functions

jianyangg committed:
Merge pull request #1 from jianyangg/branch-A-Assertions

Add assertions where appropriate to validate function arguments

jianyangg committed:
Merge pull request #2 from jianyangg/branch-A-CodeQuality

Improve code quality in accordance with the CS2103T textbook

jianyangg committed:
Add C-Priority extension to Duke

Priority is represented using exclamation marks, printed after the task name. It ranges from 0 - 5, which in turn represents the number of exclamation mark, with 0 being the least and 5 being the highest priority

jianyangg committed:
Merge pull request #3 from jianyangg/branch-C-Priority

Add C-Priority extension to Duke

jianyangg committed:
Tidy up the code quality by including comments

jianyangg committed:
Merge pull request #4 from jianyangg/branch-C-Priority

Tidy up the code quality by including comments

jianyangg committed:
Update README.md file as a User Guide

Done in accordance with CS2103T Week 6 A-UserGuide requirements

jianyangg committed:
Merge branch 'branch-A-UserGuide'

jianyangg committed:
Added Chatty name website User Guide title

jianyangg committed:
Create a JAR file for the latest Chatty version.

JAR file should not actually be included here.

It'll be released instead as v0.2

jianyangg committed:
Merge branch 'branch-A-Release'
